src/dmd/escape.d

```diff
@@ -1072,17 +1077,19 @@ bool checkNewEscape(Scope* sc, Expression e, bool gag)
          */
         if (v.storage_class & (STC.ref_ | STC.out_))
         {
-            if (global.params.useDIP25 &&
-                     sc._module && sc._module.isRoot())
+            // DIP25
+            if (sc._module && sc._module.isRoot())
             {
+                // If -preview=dip25 is used, the user wants an error
+                // Otherwise, issue a deprecation
+                const emitError = global.params.useDIP25;
                 // https://dlang.org/spec/function.html#return-ref-parameters
                 // Only look for errors if in module listed on command line
-
                 if (p == sc.func)
                 {
                     //printf("escaping reference to local ref variable %s\n", v.toChars());
                     //printf("storage class = x%llx\n", v.storage_class);
-                    escapingRef(v);
+                    escapingRef(v, emitError);
```

Comment:
It looks like all `emitError` is is a copy of `global.params.useDIP25`. The complexity of this change would be significantly reduced if you just used the global. The global is a hack anyway that will be eventually removed anyway.

Reply:
If you must use a parameter, just call it `useDIP25`, then nobody has to trace through to see that that is the only setting it is ever set to.

Reply:
The issue is that `escapingRef` is not only called for DIP25.
For example here, it is also called at L1070